src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2781,7 +2782,8 @@ private BoundExpression BindOutDeclarationArgument(DeclarationExpressionSyntax d
                         var declType = BindVariableTypeWithAnnotations(designation, diagnostics, typeSyntax, ref isConst, out isVar, out alias);
                         Debug.Assert(isVar != declType.HasType);
 
-                        return new BoundDiscardExpression(declarationExpression, declType.Type);
+                        // ValEscape is the same as for an uninitialized local
+                        return new BoundDiscardExpression(declarationExpression, Binder.ExternalScope, declType.Type);
                     }
```

Comment:
Why would this be external scope? Can you add a test like this:

```cs
Span<int> a = stackalloc int[10];
M(ref a, out _);

void M(ref Span<int> s, out Span<int> s1) => s1 = s;
```

Reply:
Also, a symmetric test like

```cs
Span<int> a = new int[10];
M(ref a, out _);

void M(ref Span<int> s, out Span<int> s1) => s1 = s;
```

would also be good to have.

Reply:
> Why would this be external scope? 

I think the added comment explains why. BTW, there is no change here for ```out``` discards. The scope used to be External and remains External.  Also, if the value is replaced with ```LocalScopeDepth``` the  Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics.RefEscapingTests.DiscardExpressionSpan unit-test is failing with unexpected errors. So, I would say we are good with test coverage for this code path.

Reply:
To be specific, the following scenarior is getting broken for example:
```
    static Span<int> Test1()
    {
        var s = ReturnsSpan(out var _);

        // ok
        return s;
    }
```

Reply:
Ah, so there's some negative downstream effects. This is interesting: we'd almost want `_` to not contribute to the scoping, either narrowing or expanding, though perhaps @jaredpar will point out some case that this wouldn't work in.